### DISCUSSION TITLE
How to initialize Prisma database after Next.js standalone build?

### DISCUSSION THREAD

**Discussion by iGmainC** [prisma/prisma#25754] [Q&A]
### Question

I am currently working on a Next.js project and deploying it in a Docker container. In the past, when I didn't use the standalone build, I would run pnpm prisma db push && next start to start the project, ensuring that the database is initialized when Docker runs for the first time (I understand that doing this when there are destructive updates to the database tables can cause data loss, but it's not a big issue during development).

Now that I am using the Next.js standalone build and placing the [.next](vscode-file://vscode-app/c:/Users/igmai/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) folder in the Docker container for deployment, I want to achieve a similar effect, i.e., initializing the Prisma database when the Docker container starts.

Do I need to write a script to execute the SQL file generated by the migration? Or is there a better way to achieve this?

Thanks a lot!

**Comment by ludralph** [comment]
Hi @iGmainC 

You can include `prisma migrate deploy` in your Docker container's entrypoint or startup script to ensure migrations are applied on container startup.
```
# Dockerfile
FROM node:18-alpine
WORKDIR /app

COPY . .

RUN pnpm install --prod
RUN pnpm prisma generate

# Copy the standalone build
COPY .next/standalone ./
COPY .next/static ./public/static

CMD ["sh", "-c", "pnpm prisma migrate deploy && node server.js"]
```

**Reply by iGmainC** [reply]
Thanks for your answers.
I will try this solution, but I have a question before trying it: there is no migration file in the `standalone/` folder. Can I also execute `pnpm prisma migrate deploy` in this case?

**Reply by ludralph** [reply]
Hi @iGmainC 

Prisma [migration](https://www.prisma.io/docs/orm/prisma-migrate/understanding-prisma-migrate/mental-model#what-is-prisma-migrate) files are typically stored in the `prisma/migrations` folder of your project, not in the `.next/standalone` directory. To run migrations in your Docker container, you need to ensure that the migration files are present. You can do this by copying the prisma directory to your Docker image. Add this line to your Dockerfile before running the migrations:
```
COPY prisma ./prisma
```
With the migration files present, you can now run `pnpm prisma migrate deploy` command. Your dockerfile should now look like:
```
FROM node:18-alpine
WORKDIR /app

COPY . .
COPY prisma ./prisma

RUN pnpm install --prod
RUN pnpm prisma generate

# Copy the standalone build
COPY .next/standalone ./
COPY .next/static ./public/static

CMD ["sh", "-c", "pnpm prisma migrate deploy && node server.js"]
```

**Comment by zaru** [comment]
@iGmainC for your reference

You should simply copy the Prisma directory.

```dockerfile
RUN npm i -g prisma # maybe necessary
COPY --from=builder --chown=nextjs:nodejs /app/prisma ./prisma/
```

I am using the official Dockerfile example as a base.
https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile
